js/blocks/controls/image.js

```diff
@@ -1,90 +1,111 @@
-const { BaseControl, TextControl, FormFileUpload, Button, Spinner } = wp.components;
+const { BaseControl, FormFileUpload, Button, Spinner } = wp.components;
+const { withSelect } = wp.data;
 const { MediaUploadCheck, MediaUpload, mediaUpload } = wp.editor;
+const { Fragment } = wp.element;
 const { __ } = wp.i18n;
 
 const BlockLabImageControl = ( props, field, block ) => {
-	const { setAttributes } = props;
-	const attr = { ...props.attributes };
-	const isUploading = 'undefined' !== typeof attr[field.name] && 'Uploading' === attr[field.name].substr( 0, 9 );
+	const ImageControl = withSelect( ( select, ownProps ) => {
+		const { attributes } = ownProps;
+		const media = select( 'core' ).getMedia( attributes[ field.name ] );
+		let imageAlt;
 
-	const uploadStart = (filename) => {
-		attr[ field.name ] = __( 'Uploading' ) + ' ' + filename;
-		setAttributes( attr )
-	};
+		// Taken from the Gutenberg Image block's edit.js file.
+		if ( media && media.alt ) {
+			imageAlt = media.alt;
+		} else if ( media && media.source_url ) {
+			imageAlt = sprintf( __( 'This image has no alt attribute, but its src is %s', 'block-lab' ), media.source_url );
+		} else {
+			imageAlt = __( 'This image has no alt attribute', 'block-lab' );
+		}
 
-	const uploadComplete = (url) => {
-		attr[ field.name ] = url;
-		setAttributes( attr )
-	};
+		return {
+			imageSrc: media ? media.source_url : '',
+			imageAlt,
+		 };
 
-	const onSelect = (image) => {
-		if ( ! image.hasOwnProperty( 'url' ) ) {
-			return
-		}
-		if ( 'blob' === image.url.substr( 0, 4 ) ) {
-			// Still uploading…
-			return
-		}
+	} )( ( ownProps) => {
+		const { imageAlt, imageSrc, setAttributes } = ownProps;
+		const attr = { ...ownProps.attributes };
+		const isUploading = 'undefined' !== typeof attr[ field.name ] && 'string' === typeof attr[ field.name ] && 'Uploading' === attr[ field.name ].substr( 0, 9 );
```

Comment:
I tried for ages to find a way of setting `isUploading` is a property of the BlockLabImageControl, so it could be set dynamically instead of just checking the value of the field. I think there's a better way here, but I don't know it. If you don't know it either, happy to commit this change.

Reply:
Sure, I'll look at this a little more. It looked pretty complex setting an `isUploading` property 😄

Reply:
https://github.com/getblocklab/block-lab/pull/260/commits/dd5ad114c17144f630a1fa4b2640f266d8d982e4 introduces a state of `isUploading`. It seems to work so far.

![uploading-image-control](https://user-images.githubusercontent.com/4063887/55473835-e37fbe80-55cc-11e9-9fd7-9a204c4d4fdf.gif)